src/cascadia/TerminalCore/TerminalSelection.cpp

```diff
@@ -395,15 +313,10 @@ COORD Terminal::_ExpandDoubleClickSelectionRight(const COORD position) const
 // - viewportPos: a coordinate on the viewport
 // Return Value:
 // - the corresponding location on the buffer
-COORD Terminal::_ConvertToBufferCell(const COORD viewportPos) const
+COORD Terminal::_ConvertToBufferCell(const COORD viewportPos) const noexcept
 {
-    // Force position to be valid
-    COORD positionWithOffsets = viewportPos;
-    _buffer->GetSize().Clamp(positionWithOffsets);
-
-    THROW_IF_FAILED(ShortSub(viewportPos.Y, gsl::narrow<SHORT>(_scrollOffset), &positionWithOffsets.Y));
-    THROW_IF_FAILED(ShortAdd(positionWithOffsets.Y, gsl::narrow<SHORT>(ViewStartIndex()), &positionWithOffsets.Y));
-    return positionWithOffsets;
+    const auto yPos = base::ClampedNumeric<short>(_VisibleStartIndex()) + viewportPos.Y;

```

Comment:
what incited the change from ViewStartIdx to VisibleStartIdx? Are they different? Why do they both exist? Why would we use one vs the otehr?

Reply:
`VisibleStartIndex`: the first VISIBLE line of the buffer (updates as we scroll)
`ViewStartIndex`: the length of the scrollback (only updates as new output forces us to scroll)

Before, the scrollback at the time of selection was saved. Then we would have to convert from "viewport position at the time of selection" to "buffer position" whenever we wanted to know what it pointed at.

Now, we have no value in knowing what the scrollback was at the time of selection. Let's just save everything as buffer positions and save the headache.

Reply:
Bingo. Thanks,